Add add_powerlaw argument to get_core_loss_eels_signal/model

magnunor opened the pull request:
The `artificial_data.get_core_loss_eels_signal` and `artificial_data.get_core_loss_eels_model` currently returns an artificial core less EEL spectrum without a power law background. This is fine for some cases, but for others (like https://github.com/hyperspy/hyperspy/pull/2003#issuecomment-403091405) having it would be nice.
 
So this pull requests add a new `add_powerlaw` argument to the `artificial_data.get_core_loss_eels_signal` and `artificial_data.get_core_loss_eels_model` functions.

### Progress of the PR
- [x] Add `add_powerlaw` argument to both functions
- [x] Update docstring
- [ ] Update user guide
- [x] add tests,
- [ ] ready for review.

### Minimal example
```python
import hyperspy.api as hs
s = hs.datasets.artificial_data.get_core_loss_eels_signal(add_powerlaw=True)
s.plot()
```

![with_powerlaw](https://user-images.githubusercontent.com/1690979/43078519-02251f10-8e8b-11e8-9c78-306c6c64a977.png)

Without the power law

```python
import hyperspy.api as hs
s = hs.datasets.artificial_data.get_core_loss_eels_signal(add_powerlaw=True)
s.plot()
```

![without_powerlaw](https://user-images.githubusercontent.com/1690979/43078563-2a5690cc-8e8b-11e8-89a6-d6d2ef408d7b.png)

magnunor committed:
Add add_powerlaw parameter to get_core_loss_eels_signal/model

magnunor commented:
I'm not really sure where to place this in the user guide, since the only place either `example_signals` or `artificial_data` is documented is in [Getting started](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/doc/user_guide/getting_started.rst#loading-example-data-and-data-from-online-databases), which I think shouldn't be filled up with a lot of extra details like this.

I guess the smart thing would be to add a new section in the user guide, about the various datasets like these which are available in HyperSpy, but that is a bit out of scope of this pull request (and my time schedule)...

thomasaarholt commented:
Would it be possible to add a "random=True" argument and code to the various artificial signals to allow them to be reproducible? I tried using them for testing puproses, but the fitted values vary slightly, which makes it less than ideal to work with for that purpose.

ericpre commented:
@thomasaarholt: you need to set the seed in your test.

ericpre reviewed (approved):
If you want to document it, you could mention it in the dev guide but other that it looks fine to me.

thomasaarholt commented:
@ericpre Thanks! I didn't realise I could just set that.

@magnunor maybe add a quick docstring reference to setting `np.random.seed(seed=some_number)` in order to make it reproducible.

magnunor committed:
docstring, get_core_loss_eels_signal: add info about setting random seed

For getting exactly the same noise every time, which can be useful
for testing fitting routines.

Also add unit test for checking this

magnunor commented:
@thomasaarholt, good idea! I added this in the docstring, and also added a unit test to check that two spectra generated using the same random seed really are exactly the same.
